### PR TITLE
Dw-155: gdpr list by subscriber

### PR DESCRIPTION
**Pull request by cbernat**
### Gdpr permission report

Add Gdpr permission fields and consent field with its value for a selected Subscriber.

image

This report will be corner stone for later implementation of a permission's historial listing.

- Add double with data for permission fields
- Add component for report
- Add route
- Add tests 

link for testing https://cdn.fromdoppler.com/doppler-webapp/int-build2188/#/reports/subscriber-gdpr?email=cbernat+c56@getcs.com

### CODE REVIEW

File: src/components/Reports/GDPR/SubscriberGdpr.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import { Loading } from '../../Loading/Loading';
+import { StarsScore } from '../../shared/StarsScore/StarsScore';
+import SafeRedirect from '../../SafeRedirect';
+
+const SubscriberGdpr = ({ location, dependencies: { dopplerApiClient } }) => {
+  const intl = useIntl();
+  const _ = (id, values) => intl.formatMessage({ id: id }, values);
+  const [state, setState] = useState({ loading: true });
+
+  const replaceSpaceWithSigns = (url) => {
```
Comment (fcoronelmakingsense) [inline src/components/Reports/GDPR/SubscriberGdpr.js]: maybe can add in utils, what do you think?

File: src/components/Reports/GDPR/SubscriberGdpr.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                              <FormattedMessage id="subscriber_gdpr.permission_description" />
+                            </th>
+                            <th scope="col">
+                              <FormattedMessage id="subscriber_gdpr.permission_value" />
+                            </th>
+                          </tr>
+                        </thead>
+                        <tbody>
+                          {state.subscriber.fields.some(
+                            (element) =>
```
Comment (fcoronelmakingsense) [inline src/components/Reports/GDPR/SubscriberGdpr.js]: need this validation if you filter after?

File: src/components/Reports/GDPR/SubscriberGdpr.test.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+          <SubscriberGdpr />
+        </IntlProvider>
+      </AppServicesProvider>,
+    );
+
+    await waitForDomChange();
+    // Assert
+    expect(getByText('subscriber_gdpr.empty_data')).toBeInTheDocument();
+  });
+});
```
Comment (fcoronelmakingsense) [inline src/components/Reports/GDPR/SubscriberGdpr.test.js]: Nice too many test! :)

File: src/services/doppler-api-client.double.ts [hunk 4]
```diff
@@ -49,6 +101,26 @@ const subscriber = {
   score: 2,
 };
 
+// const subscriberWithNoGDPR = {
```
Comment (fcoronelmakingsense) [inline src/services/doppler-api-client.double.ts]: need this comment code in double?

Comment (cbernat) [reply src/services/doppler-api-client.double.ts]: Yes because it is for testing only.

File: src/components/Reports/GDPR/SubscriberGdpr.js (hunk 2 above)
Comment (cbernat) [reply src/components/Reports/GDPR/SubscriberGdpr.js]: I want to show a message in case we have no permissions in between the fields.

Comment (cbernat) [reply src/components/Reports/GDPR/SubscriberGdpr.js]: Fixed!

File: src/components/Reports/GDPR/SubscriberGdpr.js (hunk 1 above)
Comment (cbernat) [reply src/components/Reports/GDPR/SubscriberGdpr.js]: Fixed!

**Pull request merged**